docs/fundamentals/flow/monitoring-flow.md

```diff
@@ -0,0 +1,140 @@
+
+Jina Flow expose metrics in the [prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/) which
+is plain text that is both understandable by human and machine. These metrics are intended to be scraped by
+[prometheus](https://prometheus.io/) which is an industry standard for monitoring. 
+We recommend you to visualize metrics with [grafana](https://grafana.com/)
+
+
+## Using the monitoring in a Flow
+
+A {ref}`Flow <flow-cookbook>` is composed of several Pods namely the Gateway and the Executors (potentially a Head as well). 
+see the {ref}`architecture overview <architecture-overview>` for more details. Each of these Pods are microservices which communicate 
+by gRPC. They will expose their own metrics using the [prometheus client](https://prometheus.io/docs/instrumenting/clientlibs/).
+It means that they are as many metrics endpoints that they are Pods in your Flow. 
+
+Let's give an example to illustrate it :
+
+````{tab} via Python API
+
+This example shows how to start a Flow with monitoring enabled via the Python API:
+
+```python
+from jina import Flow
+
+with Flow(monitoring=True, port_monitoring=9090).add(
+    uses='jinahub://SimpleIndexer', port_monitoring=9091
+) as f:
+    f.block()
+```
+````
+
+````{tab} via YAML
+This example shows how to start a Flow with monitoring enabled via yaml:
+
+In a `flow.yaml` file
+```yaml
+!Flow
+with:
+  monitoring: true
+  port_monitoring: 9090
+executors:
+- uses: jinahub://SimpleIndexer
+  port_monitoring: 9091
+```
+
+```bash
+jina flow --uses flow.yaml
+```
+````
+
+This Flow will create two Pods, one for the Gateway, one for the SimpleIndexer Executor, and therefore it will create two 
+metrics endpoints:
+
+* `http://localhost:9090  ` for the gateway
+* `http://localhost:9091  ` for the SimpleIndexer
+
+````{admonition} Default Monitoring port
+:class: hint
+The default monitoring port is `9090`, if you want to enable the monitoring on both the Gateway and the Executors you need to specifiy
+the `prometheus_port` for the Executors. 
+````
+
+
+Because each Pods in a Flow expose its own metrics the monitoring feature could be used independently on each Pod.
+It means that you are not forced to always monitor each Pods of your Flow. For example, you could be only interested in
+metrics coming from the Gateway, and therefore you only activate the monitoring on it. On the other hand you might be only
+interested in monitoring a single Executor. Note that by default the monitoring is disabled everywhere.
+
+To enable the monitoring you need to pass `monitoring = True` when creating the Flow.
+```python
+Flow(monitoring=True).add(...)
+```
+
+````{admonition} Enabling Flow
+:class: hint
+Passing `monitoring = True` when creating the Flow will enable the monitoring on **all the Pods** of your Flow. 
+````
+
+If you want to enable the monitoring only on the Gateway you should once you enabled it on the Flow level, disable it for 
+all the other Executors.
```

Comment:
Above there is an explanation saying that it is possible to monitor only certain microservices, but the explanation how to do it is only here, after the explanation of how to enable monitoring for the entire flow. I think this ordering might be more natural: 1. Say that you can enable monitoring for the entire flow 2. show how to do that 3. say that you can also monitor only select services 4. show how to do that

```suggestion
If you want to enable the monitoring only on the Gateway, you need to first enable the feature for then entire Flow, and then disable it for the Executor which you are not interested in.
```